pyramid/directory.go

```diff
@@ -27,23 +27,19 @@ func (d *directory) deleteDirRecIfEmpty(dir string) error {
 
 	for dir != d.ceilingDir {
 		empty, err := isDirEmpty(dir)
-		if err != nil {
-			if errors.Is(err, os.ErrNotExist) {
-				return nil
-			}
-			return err
-		}
-		if !empty {
+		if errors.Is(err, os.ErrNotExist) {
 			return nil
 		}
+		if err != nil || !empty {
```

Comment:
It's not that it's not valid, but it's somewhat hard to understand. Maybe simplify with:
```suggestion
		if err != nil {
		   return err
		 }
		 if !empty {
		   return nil
		 }
```

Reply:
that was the previous version I had :) thought the shorter version made mode sense